src/playwright/test.ts

```diff
@@ -76,6 +77,153 @@ export const expect = baseExpect.extend({
     };
   },
 
+  async toDispatchEvents(
+    locator: Locator,
+    action: () => Promise<unknown>,
+    expectedEvents: {
+      bubbles?: boolean;
+      cancelable?: boolean;
+      composed?: boolean;
+      defaultPrevented?: boolean;
```

Comment:
I, unfortunately, had to make every key but `type` optional for cases like [this](https://github.com/CrowdStrike/glide-core/pull/1036/files#diff-53297a310bb23c9c791ccfe4ebdda9c0235aacae3df20e29571f8a7fb32c8c7fR41-R48).